src/Compiler/TypedTree/TypedTreeBasics.fs

```diff
@@ -245,7 +245,11 @@ let rec stripUnitEqnsAux canShortcut unt =
 let combineNullness (nullnessOrig: Nullness) (nullnessNew: Nullness) = 
     match nullnessOrig.Evaluate() with
     | NullnessInfo.WithoutNull -> nullnessNew
-    | NullnessInfo.AmbivalentToNull -> nullnessOrig
+    | NullnessInfo.AmbivalentToNull ->
+        match nullnessNew.Evaluate() with
+        | NullnessInfo.WithoutNull -> nullnessOrig
+        | NullnessInfo.AmbivalentToNull -> nullnessOrig
+        | NullnessInfo.WithNull -> nullnessNew
```

Comment:
The combination of orig=Ambivalent and New=WithNull is what was happening in a reported bug.

The situation was with DirectoryInfo (extends FileSystemInfo):

let d : DirectoryInfo | null = null
d.Name  // .Name is a property from the super type


No warning was raised.
The reason was, the .Name access combined the ambivalent nullness of the supertype info (as imported from IL metadata) and the WithNull nullness of the `d` value.